src/main/java/ui/ChatBot.java

```diff
@@ -0,0 +1,57 @@
+package ui;
+
+import data.TaskList;
+import ui.command.Command;
+
+import java.util.ArrayList;
+
+/**
+ * ChatBot class that allows for running of user command
+ * and keeping track of its termination status.
+ */
+public class ChatBot {
+
+    private static final String WELCOME_STRING = "Hello I'm Duke!\n"
+            + "What can I do for you?";
+
+    /** Task list maintaining list of tasks for user */
+    private final TaskList taskList;
+
+    /** Boolean to track if bot has received a termination command */
+    private boolean hasTerminated;
+
+    public ChatBot(TaskList taskList) {
+        this.taskList = taskList;
+        this.hasTerminated = false;
+    }
+
+
+    /**
+     * Performs initialisation of the ChatBot. Should be called
+     * before receiving commands.
+     */
+    public void initialise() {
+        System.out.println(WELCOME_STRING);
+    }
+
+    /**
+     * Runs the command given by user input string.
+     *
+     * @param input String command by user to run.
+     */
+    public void runCommand(String input) {
+        try {
+            Command command = Command.parseCommand(input, this.taskList);
+            this.hasTerminated = command.execute();
+        } catch (IllegalArgumentException e) {
+            ArrayList<String> response = new ArrayList<>();
```

Comment:
Thanks for pointing this out! **responses** might not be appropriate since it is a single response by my ChatBot to a command, but I do agree that I should change it to reflect its state as a collection.